pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -390,45 +372,110 @@ func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACL(ctx context.Contex
 		log.Error(msg)
 		return errors.New(msg)
 	}
-	if len(clientVms) == 0 {
-		// Prepare the CnsVolumeACLConfigureSpec
-		cnsVolumeID := cnstypes.CnsVolumeId{
-			Id: volumeID,
+	if !delete && len(clientVms) == 0 {
+		err = r.configureVolumeACLs(ctx, volumeID, tkgVMIP, false)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to add net permissions for file volume %q. Error: %+v", volumeID, err)
+			log.Error(msg)
+			return errors.New(msg)
 		}
-		vSanFileShareNetPermissions := make([]vsanfstypes.VsanFileShareNetPermission, 0)
-		vsanFileShareAccessType := vsanfstypes.VsanFileShareAccessTypeREAD_WRITE
-		vSanFileShareNetPermissions = append(vSanFileShareNetPermissions, vsanfstypes.VsanFileShareNetPermission{
-			Ips:         tkgVMIP,
-			Permissions: vsanFileShareAccessType,
-		})
-
-		cnsNFSAccessControlSpecList := make([]cnstypes.CnsNFSAccessControlSpec, 0)
-		cnsNFSAccessControlSpecList = append(cnsNFSAccessControlSpecList, cnstypes.CnsNFSAccessControlSpec{
-			Permission: vSanFileShareNetPermissions,
-		})
-
-		cnsVolumeACLConfigSpec := cnstypes.CnsVolumeACLConfigureSpec{
-			VolumeId:              cnsVolumeID,
-			AccessControlSpecList: cnsNFSAccessControlSpecList,
+	}
+	if delete && len(clientVms) == 1 && clientVms[0] == vm.Name {
```

Comment:
In conjunction with my previous comment, how about changing this to `else` and then within that `if len(clientVms) == 1 && clientVms[0] == vm.Name`. This way you can avoid the redundant `if` on line 391 by bringing the code in this path

Reply:
Addressed in the latest commit